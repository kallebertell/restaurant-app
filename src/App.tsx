import * as React from 'react';
import { connect } from 'react-redux';
import { Route, Switch } from 'react-router';
import { Action, ActionCreator } from 'redux';
import styled from 'styled-components';

import { fetchAuthRequest } from 'store/auth/authActions';
import { getAuthToken } from 'store/auth/authSelectors';
import { AppState } from 'store/rootReducer';

const PageHome = () => <div>Welcome to pizza</div>
const PageRestaurantListing = () => <div>Here we have restaurants</div>
const PageRestaurantDetails = () => <div>Here's a specific restaurant</div>
const PageNotFound = () => <div>Page not Found</div>;

const AppWrapper = styled.div`
  text-align: center;
`
const Header = styled.header`
  background: black;
  color: white;
  padding: 1rem;
`
const Title = styled.h1`
  text-transform: uppercase;
  margin: 0;
`;

interface Props {
  token?: string;
  fetchAuthRequest: ActionCreator<Action>;
}

export class App extends React.Component<Props> {
  public componentDidMount() {
    if (!this.props.token) {
      this.props.fetchAuthRequest();
    }
  }

  public render() {
    return (
      <AppWrapper>
        <Header>
          <Title>Eat Pizza</Title>
        </Header>
        <Switch>
          <Route exact={true} path="/" component={PageHome}/>
          <Route exact={true} path="/restaurants/:city/:postCode" component={PageRestaurantListing}/>
          <Route exact={true} path="/restaurant/:id" component={PageRestaurantDetails}/>
          <Route path="*" component={PageNotFound} />
        </Switch>
      </AppWrapper>
    );
  }
}

const mapStateToProps = (state: AppState) => ({
  token: getAuthToken(state)
});

export default connect(
  mapStateToProps,
  { fetchAuthRequest }
)(App);
