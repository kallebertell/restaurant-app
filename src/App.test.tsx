import * as React from 'react';

import { shallow } from 'enzyme';

import {Â App } from './App';

const fetchAuthRequest = () => ({ type: 'action' });

it('renders without crashing', () => {
  const wrapper = shallow(<App fetchAuthRequest={fetchAuthRequest}/>);
  expect(wrapper).toMatchSnapshot();
});
