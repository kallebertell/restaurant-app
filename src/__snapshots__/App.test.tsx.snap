// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<styled.div>
  <styled.header>
    <styled.h1>
      Eat Pizza
    </styled.h1>
  </styled.header>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/restaurants/:city/:postCode"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/restaurant/:id"
    />
    <Route
      component={[Function]}
      path="*"
    />
  </Switch>
</styled.div>
`;
